collection.c

```diff
@@ -1622,7 +1631,11 @@ static void php_mongocollection_remove(zval *this_ptr, mongo_collection *c, zval
 		RETURN_FALSE;
 	}
 
-	if (php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_WRITE_API)) {
+	is_acknowledged = is_gle_op(this_ptr, z_write_options, &link->servers->options TSRMLS_CC);
+	supports_command = php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_WRITE_API);
+	supports_opcode = php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_RELEASE_2_4_AND_BEFORE);
```

Comment:
As you're doing this 3 times - would it make sense to split it up in a method call with three "out" variables? Something like:

```
static void figure_out_support(this_ptr, z_write_options, link->server->options, connection, &is_acknowledged, &supports_command, &supports_opcode);
```